src/extensions/__tests__/is-compatible-extension.test.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import semver from "semver";
+import {
+  isCompatibleExtension,
+} from "../extension-discovery/is-compatible-extension/is-compatible-extension";
+import type { LensExtensionManifest } from "../lens-extension";
+
+describe("Extension/App versions compatibility check", () => {
```

Comment:
I like we should force extensions to not be compatible if they try and use a non-`^` range, can we have a few tests to that affect?

Reply:
> try and use a non-^ range

do you mean like  this format too`>=2.1.0`?

Reply:
Yeah we don't really want an extension to be able to specify that they support multiple major versions at a time. Plus we don't want to support extensions saying `~` ranges